Shared/Samples/Sketch on the map/SketchOnTheMapView.swift

```diff
@@ -0,0 +1,289 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+/// A view that shows how to interact with the geometry editor.
+struct SketchOnTheMapView: View {
+    /// The map to display in the view.
+    @State private var map = Map(basemapStyle: .arcGISTopographic)
+    /// The model that is required by the menu.
+    @StateObject var model = GeometryEditorMenuModel(
+        geometryEditor: GeometryEditor(),
+        graphicsOverlay: GraphicsOverlay(renderingMode: .dynamic)
+    )
+    
+    var body: some View {
+        VStack {
+            MapView(map: map, graphicsOverlays: [model.graphicsOverlay])
+                .geometryEditor(model.geometryEditor)
+        }
+        .toolbar {
+            ToolbarItem(placement: .primaryAction) {
+                GeometryEditorMenu(model: model)
+            }
+        }
+    }
+}
+
+/// A view that provides a menu for geometry editor functionality.
+struct GeometryEditorMenu: View {
+    /// The model for the menu.
+    @ObservedObject var model: GeometryEditorMenuModel
+    
+    var body: some View {
+        Menu {
+            if !model.isStarted {
+                // If the geometry editor is not started, show the main menu.
+                mainMenuContent
+            } else {
+                // If the geometry editor is started, show the edit menu.
+                editMenuContent
+            }
+        } label: {
+            Label("Geometry Editor", systemImage: "pencil.tip.crop.circle")
+        }
+    }
+}
+
+extension GeometryEditorMenu {
+    /// The content of the main menu.
+    private var mainMenuContent: some View {
+        VStack {
+            Button {
+                model.geometryEditor.tool = VertexTool()
+                model.geometryEditor.start(withType: Point.self)
+                model.isStarted = true
+            } label: {
+                Label("New Point", systemImage: "pencil")
+            }
+            
+            Button {
+                model.geometryEditor.tool = VertexTool()
+                model.geometryEditor.start(withType: Polyline.self)
+                model.isStarted = true
+            } label: {
+                Label("New Line", systemImage: "pencil.line")
```

Comment:
For some reason, `pencil.line` doesn't render for me. 🤔 

<img width="394" alt="Screenshot 2023-03-22 at 17 17 40" src="https://user-images.githubusercontent.com/16397058/227067212-ec1cce84-c1c9-4102-825a-e8e4bc26a5e8.png">

Reply:
I would recommend cleaning, deleting derived data and trying again.

Can you try some other symbols and see if they exist?

I probably wouldn't hold up the PR for this.

Reply:
<img width="332" alt="image" src="https://user-images.githubusercontent.com/9660181/227298443-66d33d75-590a-4d2f-9e79-c1e3f144038a.png">

I'm guessing it is related to SF Symbols versions. On iPhone 14 Pro Max iOS 16.2 sim I see it display, whereas it doesn't show on iPhone 11 iOS 15.5. Maybe it is a new symbol shipped with a later version of iOS.

Reply:
So if we target iOS 15.0, an alternative can be `line.diagonal`

Reply:
Yup, I think this is correct, I'm seeing the same.